### PR TITLE
Support for defining optional bearerFormat.

### LINKED ISSUE #2038: [FEATURE] Ability to define `bearerFormat` in OpenAPI documentation
OpenAPI specification supports an optional parameter called `bearerFormat` in security scheme.

See for instance https://swagger.io/docs/specification/authentication/bearer-authentication/:

```yaml
openapi: 3.0.0
...
# 1) Define the security scheme type (HTTP bearer)
components:
  securitySchemes:
    bearerAuth:            # arbitrary name for the security scheme
      type: http
      scheme: bearer
      bearerFormat: JWT    # optional, arbitrary value for documentation purposes
# 2) Apply the security globally to all operations
security:
  - bearerAuth: []         # use the same name as above
```

It would be nice if Tapir supported defining this `bearerFormat`, typically on the `TapirAuth#bearer()` method.

### PR DESCRIPTION
**Pull request by marcin-jozefowicz**
closes #2038

### CODE REVIEW

**Comment by marcin-jozefowicz** [comment]
According to documentation:

**Comment by adamw** [comment]
Looks good :) I don't think it makes sense to go to great lengths just to make sure somebody doesn't use this incorrectly :)

File: core/src/main/scala/sttp/tapir/EndpointIO.scala [hunk 1]
```diff
@@ -218,6 +218,8 @@ object EndpointInput extends EndpointInputMacros {
 
     def description(d: String): Auth[T, TYPE] = copy(info = info.description(d))
 
+    def bearerFormat(f: String): Auth[T, TYPE] = copy(info = info.bearerFormat(f))
```
Comment (adamw) [inline core/src/main/scala/sttp/tapir/EndpointIO.scala:221]: one thing we might do is add a constraint here on `TYPE` to ensure that we're calling it on AuthType.Http only, using sth like `(implicit typeIsHttp: TYPE =:= AuthType.Http)`

**Comment by gaeljw** [comment]
Looks good :) Thanks!

**Pull request merged**